### PR TITLE
Update DNA Distribution Code

### PR DESCRIPTION
**Pull request by CarterZenke**

### CODE REVIEW

File: dna.py [hunk 1]
```diff
@@ -0,0 +1,51 @@
+import csv
+import sys
+
+def main():
+    # TODO: Check for proper usage
```
Comment (dmalan) [inline dna.py:5]: Let's add a blank line above first comment. And change comment to "Check for proper command-line usage".

File: dna.py [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+
+    # TODO: Read database file and DNA sequence file
```
Comment (dmalan) [inline dna.py:7]: Separate into two separate TODOs? And perhaps say "into a variable" for each?

File: dna.py [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+
+    # Initialize variables
+    longest_run = 0
+    subseq_length = len(subsequence)
+    seq_length = len(sequence)
+
+    # Check each character in sequence for most consecutive runs of subsequence
+    for i in range(seq_length):
+
+        count = 0
```
Comment (dmalan) [inline dna.py:26]: Let's comment what this represents?

File: dna.py [hunk 4]
```diff
@@ -0,0 +1,51 @@
...
+    # TODO: Check database for matching profiles
+
+    return
+
+def longest_match(sequence, subsequence):
+    """Returns length of longest run of subsequence in sequence."""
+
+    # Initialize variables
+    longest_run = 0
+    subseq_length = len(subsequence)
```
Comment (dmalan) [inline dna.py:20]: Let's do `subsequence_length` for clarity?

File: dna.py [hunk 5]
```diff
@@ -0,0 +1,51 @@
...
+    seq_length = len(sequence)
```
Comment (dmalan) [inline dna.py:21]: `sequence_length`

File: dna.py [hunk 6]
```diff
@@ -0,0 +1,51 @@
...
+
+    # Check each character in sequence for most consecutive runs of subsequence
+    for i in range(seq_length):
+
+        count = 0
+
+        # Check for a match and continue checking until out of consecutive matches
+        while True:
+
+            # Adjust window start and end
```
Comment (dmalan) [inline dna.py:31]: windows'

But what does "window" mean here? We should define in this or an earlier comment.

File: dna.py [hunk 7]
```diff
@@ -0,0 +1,51 @@
...
+            # If there is no match in the window
+            else:
+                break
+        
+        # Update most consecutive matches found
+        longest_run = max(longest_run, count)
+
+    # After checking for runs at each character in seqeuence, return longest run found
+    return longest_run
+
```
Comment (dmalan) [inline dna.py:48]: 2 blank lines per PEP I think

File: dna.py [hunk 8]
```diff
@@ -0,0 +1,51 @@
...
+# When dna.py is run directly, execute main
+if __name__ == "__main__":
```
Comment (dmalan) [inline dna.py:50]: Do we need the `if`? Is `check50` importing this file for testing? If not, let's just use `main()` on its own?

File: dna.py (hunk 6 above)
Comment (CarterZenke) [reply dna.py:31]: I changed to "substring" since that language seems more consistent with what we've used in the problem set specifications.

**Pull request merged**